File: fsharp/grains/Grains.fs
module Grains

let square (n: int): Result<uint64,string> =
    match n with
    | _ when 1 <= n && n <= 64 -> 
        Ok (1UL <<< (n - 1))
    | _ -> 
        Error "square must be between 1 and 64"

let total: Result<uint64,string> = 
    [1..64] 
    |> List.map square 
    |> List.reduce (fun a b -> 
        match a, b with
        | Ok x, Ok y -> Ok (x + y)
        | _ -> Error "Invalid input")

File: go/grains/grains.go
// Package grains provides methods helpful in the booming
// field of Applied Grain Mathematics
package grains

import "errors"

// Square returns the number of grains on a given square
func Square(n int) (uint64, error) {
	if n < 1 || n > 64 {
		return 0, errors.New("square number out of range")
	}
	return 1 << (n - 1), nil
}

// Total returns the total number of grains on the chessboard
func Total() uint64 {
	return 1<<(65-1) - 1
}
